system-metrics/system-metrics.gradle

```diff
@@ -0,0 +1,11 @@
+apply from: "$rootDir/gradle/java.gradle"
+apply from: "$rootDir/gradle/publish.gradle"
+
+
+dependencies {
+  implementation deps.opentelemetryApi
+  implementation "com.github.oshi:oshi-core:5.2.5"
```

Comment:
An alternative to this would be to add this but:

1) Disable it by default AND
2) Make oshi a compile dependency.

This way, if users want to leverage it, they'd need to explicitly enable this instrumentation part, AND provide the Oshi dependency themselves.

Reply:
I think it's a good idea to make oshi a compile dependency. How about we model this change exactly like any instrumentation, `instrumentation/oshi/{library,javaagent}` for example. This code goes in library, giving user a manual way of instrumenting (calling the static method) similar to our other library instrumentations, and auto can call it automatically if oshi is found on the classpath (not sure if we have a pattern for this yet but figure it's possible right @trask?). We can start with just moving this into library and add javaagent later.

For reference, https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/master/instrumentation/grpc-1.5/library/grpc-1.5-library.gradle

Reply:
In which case we should split out the JVM metrics into a separate `instrumentation/jvm/library` too

Reply:
> This code goes in library, giving user a manual way of instrumenting (calling the static method) similar to our other library instrumentations

Sounds good to me.

> In which case we should split out the JVM metrics into a separate instrumentation/jvm/library too

Same.

@malafeev any opinion on this? ;)

Reply:
I'm fine with it. any other opinions?

Reply:
I like the idea of adding just library instrumentation in the first PR, and then work on how to integrate it into javaagent after that

Reply:
I moved code to `instrumentation/oshi/library` because both jvm and system metrics depends on oshi